tools/graph-convert/graph-properties-convert.cpp

```
@@ -0,0 +1,1852 @@
+    break;
+  }
+  }
+  if (!st.ok()) {
+    GALOIS_LOG_FATAL("Error adding value to arrow array builder");
+  }
+}
+
+void convertToPropertyGraphAndWrite(const GraphComponents& graphComps,
+                                    const std::string& dir) {
+  galois::graphs::PropertyFileGraph graph;
+
+  auto result = graph.SetTopology(*graphComps.topology);
+  if (!result) {
+    GALOIS_LOG_FATAL("Error adding topology");
+  }
+
+  if (graphComps.nodeProperties->num_columns() > 0) {
+    result = graph.AddNodeProperties(graphComps.nodeProperties);
+    if (!result) {
+      GALOIS_LOG_FATAL("Error adding node properties");
+    }
+  }
+  if (graphComps.nodeLabels->num_columns() > 0) {
+    result = graph.AddNodeProperties(graphComps.nodeLabels);
+    if (!result) {
+      GALOIS_LOG_FATAL("Error adding node labels");
+    }
+  }
+  if (graphComps.edgeProperties->num_columns() > 0) {
+    result = graph.AddEdgeProperties(graphComps.edgeProperties);
+    if (!result) {
+      GALOIS_LOG_FATAL("Error adding edge properties");
+    }
+  }
+  if (graphComps.edgeTypes->num_columns() > 0) {
+    result = graph.AddEdgeProperties(graphComps.edgeTypes);
+    if (!result) {
+      GALOIS_LOG_FATAL("Error adding edge types");
+    }
+  }
+
+  tsuba::Init();
+  std::string metaFile = dir + "/meta";
+  result               = graph.Write(metaFile);
+  if (!result) {
+    GALOIS_LOG_FATAL("Error writing to fs");
+  }
+  tsuba::Fini();
+}
+
+template <typename T>
+std::vector<T> parseNumberList(std::string rawList) {
+  std::vector<T> list;
+
+  if (rawList.front() == '[' && rawList.back() == ']') {
+    rawList.erase(0, 1);
+    rawList.erase(rawList.length() - 1, 1);
+  } else {
+    GALOIS_LOG_ERROR("The provided list was not properly formatted");
+    return list; // TODO throw better error
+  }
+  std::vector<std::string> elems;
+  boost::split(elems, rawList, boost::is_any_of(","));
+
+  for (std::string s : elems) {
+    std::istringstream iss(s);
+    T elem;
+    iss >> elem;
```

this will silently fail if s cannot be parsed into `elem` . Perhaps, if (!(iss >> elem)) { ... } else { ... }`?